extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java

```diff
@@ -557,6 +563,53 @@ private boolean isHibernateProvidedBasicType(DotName dotName) {
         return DotNames.HIBERNATE_PROVIDED_BASIC_TYPES.contains(dotName);
     }
 
+    // Tries to get the parent (name) of a field that is typed, e.g.:
+    // class Foo<ID extends Serializable> { @EmbeddedId ID id; }
+    // class Bar<ID extends SomeMoreSpecificBaseId> extends FOO<ID> { }
+    // class Baz extends Bar<BazId> { }
+    // @Embeddable class BazId extends SomeMoreSpecificBaseId { @Column String a; @Column String b; }
+    //
+    // Without this method, type of Foo.id would be Serializable and therefore Foo.id.a (or b) would not be found.
+    //
+    // Note: This method is lenient in that it doesn't throw exceptions aggressively when an assumption is not met.
+    private DotName getParentNameFromTypedFieldViaHierarchy(FieldInfo fieldInfo, List<ClassInfo> parentSuperClassInfos) {
```

Comment:
This is the second, more complex issue I mentioned. Without this you'll get the following IT failure:
```
[ERROR] Errors:
[ERROR]   PhoneCallResourceTest.testFindByAreaCode » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.spring.data.deployment.SpringDataJPAProcessor#build threw an exception: java.lang.IllegalStateException: Entity class java.io.Serializable referenced by io.quarkus.it.spring.data.jpa.PhoneCall.id was not part of the Quarkus index. Offending method is 'findByIdAreaCode' of repository 'io.quarkus.it.spring.data.jpa.PhoneCallRepository'.
```
The reason for this slightly irritating error is that `fieldInfo.type().name()` for the type `ID extends Serializable` will just return `Serializable` .